### PR TITLE
enh (api/command): Create new API endpoint to list all commands

### PR DESCRIPTION
**Pull request by callapa**
## Description

Create new API endpoint to list all commands
```
GET /configuration/commands
```

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Command/Infrastructure/Repository/DbReadCommandRepository.php [hunk 1]
```diff
@@ -79,9 +104,78 @@ public function existsByIdAndCommandType(int $commandId, CommandType $commandTyp
...
+        $request = <<<'SQL'
+            SELECT
+                command_id,
+                command_name,
+                command_line,
+                command_type,
+                enable_shell,
+                command_activate,
+                command_locked
+            FROM `:db`.command
```
Comment (callapa) [inline centreon/src/Core/Command/Infrastructure/Repository/DbReadCommandRepository.php:142]: must to be filtered according to the command types.

File: centreon/src/Core/Command/Infrastructure/Model/CommandTypeConverter.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            CommandType::Check => 2,
+            CommandType::Miscellaneous => 3,
+            CommandType::Discovery => 4,
+        };
+    }
+
+    /**
+     * @param int $commandType
+     *
+     * @return CommandType
```
Comment (callapa) [inline centreon/src/Core/Command/Infrastructure/Model/CommandTypeConverter.php:48]: Add PHPDOC @throws \RangeException

File: centreon/src/Core/Command/Application/UseCase/FindCommands/FindCommandsResponse.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Command\Application\UseCase\FindCommands;
+
+class FindCommandsResponse
```
Comment (abuathier) [inline centreon/src/Core/Command/Application/UseCase/FindCommands/FindCommandsResponse.php:26]: phpstan `Class FindCommandsResponse must be final.`
```suggestion
final class FindCommandsResponse
```

**Review by abuathier — APPROVED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

File: centreon/tests/php/Core/Command/Application/UseCase/FindCommands/FindCommandsTest.php [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    ($this->useCase)($this->presenter);
+
+    expect($this->presenter->response)
+        ->toBeInstanceOf(ForbiddenResponse::class)
+        ->and($this->presenter->response->getMessage())
+        ->toBe(CommandException::accessNotAllowed()->getMessage());
+});
+
+it(
+    'should present a ErrorResponse when an exception of type RequestParametersTranslatorException is thrown',
```
Comment (cg-tw) [inline centreon/tests/php/Core/Command/Application/UseCase/FindCommands/FindCommandsTest.php:83]:
```suggestion
    'should present an ErrorResponse when an exception of type RequestParametersTranslatorException is thrown',
```

**Review by cg-tw — DISMISSED** [review]

**Review by dmyios — DISMISSED** [review]

**Review by abuathier — APPROVED** [review]

**Review by sc979 — APPROVED** [review]
LGTM on veracode part

**Review by dmyios — APPROVED** [review]

**Review by abuathier — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**